### PR TITLE
Fix "find and replace" initializing with wrong size

### LINKED ISSUE #14812: Code editor goes outside its bounds
**Godot version:**
master: d81bd96

**Issue description:**
I first made an [issue](https://github.com/godotengine/godot/issues/12012) thinking that the problem was with the search/replace bar, but it happens even with without them. The VSplitContainer should stop dragging once it has reached the limit, but it just keeps going until it's outside the bounds of it.

bug

### PR DESCRIPTION
**Pull request by guilhermefelipecgs**
Regression.

captura de tela de 2018-06-10 19-13-42

Update:
Added fix #14812.
out mp4

### CODE REVIEW

File: editor/code_editor.cpp [hunk 1]
```diff
@@ -481,6 +481,9 @@ void FindReplaceBar::set_text_edit(TextEdit *p_text_edit) {
 void FindReplaceBar::_update_size() {
 
 	container->set_size(Size2(hbc->get_size().width, 1));
+
+	// update variable tab_container in script_editor_plugin
+	get_parent_control()->get_parent_control()->get_parent_control()->minimum_size_changed();
```
Comment (akien-mga) [inline editor/code_editor.cpp]: Such an hardcoded call to the grand-grandparent of the node doesn't seem like the right fix. Usually such changes are propagated with signals.

Comment (guilhermefelipecgs) [reply editor/code_editor.cpp]: Yeah, I agree that this is weird but to emit a signal, if I remember well, I'll have to create an signal on FindReplaceBar, CodeTextEditor, ScriptTextEditor to finally connect it on ScriptEditorPlugin to update tab_container. Unless there is a better way of propagating that I haven't realized.

I'll give a check on it.

**Comment by akien-mga** [comment]
We discussed this on IRC, @reduz and @groud mentioned that the anchoring code should be removed now that the script editor is made a VBoxContainer, and that should remove the need for the hackish call to the grandparent's `minimum_size_changed`.

**Comment by guilhermefelipecgs** [comment]
Removing the anchoring will work more or less, this hack fix that particular situation:
out5 mp4
When the FindReplaceBar appear, the tab_container only update when a `NOTIFICATION_RESIZED` occurs.

If you don't mind with that I can remove and let this way or I can do the "signals changes".

**Comment by groud** [comment]
Then the problem is probably a little bit deeper than that. The HBoxContainer should emit a resized signal when it's content is made visible. That's probably where the problem lies.

**Comment by guilhermefelipecgs** [comment]
@groud Agree, but I notice that it's doesn't happen, if we expect this behaviour then we have another bug. I'll look this asap.

**Comment by guilhermefelipecgs** [comment]
Ok, I found the problem, `TabContainer` was not a `Control`.

**Comment by groud** [comment]
> Ok, I found the problem, TabContainer was not a Container.

That was quite unexpected... The change makes sense I guess.

**Comment by guilhermefelipecgs** [comment]
Yeah, I'm surprised too. :)

**Comment by guilhermefelipecgs** [comment]
Don't merge yet, I'll make some tests.

**Comment by guilhermefelipecgs** [comment]
I'll check for possible regressions here.

**Comment by fire** [comment]
This regressed visual scripting editor.

**Comment by fire** [comment]
```patch
Subject: [PATCH] Fix regression from "Fix "find and replace" initializing with
 wrong size".

---
 modules/visual_script/visual_script_editor.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/visual_script/visual_script_editor.cpp b/modules/visual_script/visual_script_editor.cpp
index 6df0b5036b5..532d79423cb 100644
--- a/modules/visual_script/visual_script_editor.cpp
+++ b/modules/visual_script/visual_script_editor.cpp
@@ -3481,6 +3481,7 @@ VisualScriptEditor::VisualScriptEditor() {
 
 	graph = memnew(GraphEdit);
 	add_child(graph);
+	graph->set_v_size_flags(Control::SIZE_EXPAND_FILL);
 	graph->set_anchors_and_margins_preset(Control::PRESET_WIDE);
 	graph->connect("node_selected", this, "_node_selected");
 	graph->connect("_begin_node_move", this, "_b
```

**Comment by guilhermefelipecgs** [comment]
Thanks.

**Pull request merged**